source/esbuild-plugin.civet

```diff
@@ -13,16 +15,91 @@ esbuild.build({
   ]
 }).catch(() => process.exit(1))
 ```
+
+Chain civet output into another esbuild plugin, solid for example
+
+@example
+```javascript
+import esbuild from 'esbuild'
+const { solidPlugin } = require('esbuild-plugin-solid');
+import civetPlugin from '@danielx/civet/esbuild-plugin'
+
+esbuild.build({
+  ...,
+  plugins: [
+    civetPlugin(
+      next: solidPlugin()
+    )
+  ]
+}).catch(() => process.exit(1))
+```
 */
 
-{ readFile } := require 'fs/promises'
+{ readFile, writeFile } := require 'fs/promises'
 path := require 'path'
 
 // NOTE: this references the built version of the module, not the source
 { compile } := require "../dist/main.js"
 
-module.exports = {
-  name: 'civet'
+interface Options
+  filter?: RegExp
+  inlineMap?: boolean
+  js?: boolean
+  next?: unknown
```

Comment:
I'm not very familiar with the esbuild plugin system, but I wonder if it makes sense to support multiple `next` transforms, as an array.  I needed this for the analogous option in the Vite plugin (see https://github.com/lorefnon/vite-plugin-civet/pull/5) to e.g. process JSX and then add compile-time constants.  But if most plugins already support a `next` option, it's probably not necessary...

Reply:
I could be possible though I don't know of any uses off the top of my head but I'm sure some might come up. Most esbuild plugins are unchainable by default. Maybe even extracting this tmp file technique to arbitrarily chain esbuild plugins could be useful more generally than civet.

Reply:
We can always leave this for later when a need arises, as it did for the Vite plugin.